shell/platform/android/io/flutter/embedding/legacy/PluginRegistry.java

```diff
@@ -107,13 +108,6 @@
          */
         PlatformViewRegistry platformViewRegistry();
 
-        /**
-         * Returns the {@link FlutterView} that's instantiated by this plugin's
-         * {@link #activity() activity}.
-         */
-        FlutterView view();
```

Comment:
Is there no reason for plugins to need the FlutterView anymore? I would assume this is going to break some plugins...
(I realize there might not be a FlutterView, maybe we can return an Optional, or a null if there's no view?)

Reply:
I think it would help to put together a short list of very diverse plugin behaviors that could help demonstrate requirements.  Would you mind helping with that?  For example, you had a good point earlier about the video plugin.

Whatever the use-cases, I think we need to introduce "Android View attachment" as part of the plugin lifecycle.  I don't think providing a reference via "view()" is sufficient, we should give plugins an opportunity to setup and teardown when the Android View comes and goes.  Moreover, if we introduce lifecycle callbacks then we can pass the FlutterView reference in those callbacks instead of giving the impression that it is always available via "view()".

Reply:
I'm going to mark this conversation resolved for this PR because it will be dealt with in future PRs and requires deeper discussion.